src/compiler/crystal/semantic/type_declaration_processor.cr

```diff
@@ -269,7 +269,7 @@ struct Crystal::TypeDeclarationProcessor
 
       # Reject annotations to existing instance var
       type_decl.annotations.try &.each do |_, ann|
-        ann.raise "can't annotate an instance variable not defined in this class"
+        ann.raise "can't annotate #{name} here because it was first defined in #{supervar.owner}"
```

Comment:
```suggestion
        ann.raise "can't annotate #{name} in #{owner} because it was first defined in #{supervar.owner}"
```

Reply:
Sounds good. I'll add the commit myself to fix the spec.